tagsbox/react-instantsearch/README.md

```diff
@@ -0,0 +1,118 @@
+  * [CSS](#css)
+* [Usage](#usage)
+  * [Requirements](#requirements)
+* [API](#api)
+  * [relevanceWidget(opts)](#relevancewidgetopts)
+* [Examples](#examples)
+* [Implementation details](#implementation-details)
+* [Contributing](#contributing)
+
+## Get the code
+
+This widget comes with JavaScript but also pre-defined CSS.
+
+### JavaScript
+
+You can copy and paste the JavaScript code from the repository itself, grab it in the [src/lib](src/lib) folder and use the `<Tag />` component.
+
+### CSS
+
+You can copy and paste the necessary CSS code from the repository itself, grab it in [src/lib/Tags.css](src/lib/Tags.css).
+
+## Usage
+
+The simplest usage is:
+
+```jsx
+import { InstantSearch } from 'react-instantsearch-dom';
+import TagsBox from 'instansearch-tagsbox-react';
+
+export const TagSelectedComponent = ({ hit }) => (
+    <Fragment>
+        {hit.lastname}
+    </Fragment>
+);
+
+export const TagSuggestionComponent = ({ hit }) => (
+    <Fragment>
+        {hit.firstname} {hit.lastname}
+    </Fragment>
+);
+
+<InstantSearch
+    appId="..."
+    apiKey="..."
+    indexName="..."
+>
+    <TagsBox
+        tagSelectedComponent={TagSelectedComponent}
+        tagSuggestionComponent={TagSuggestionComponent}
+        onUpdate={(newTags, oldTags) => console.log}
+        translations={{ placeholder: "Search…", noResult: "…" }}
+    />
+</InstantSearch>
+```
+
+### Requirements
+
+You should encapsulte the `<TagsBox />` component inside an `<InstantSearch />` one.
+
+To work properly, **you'll need to provide three props to the component**:
+
+- `tagSelectedComponent` (component): describe how a selected tag should be displayed
+  - receive a `hit` as parameter (coming from Algolia)
+- `tagSuggestionComponent` (component): describe how a suggested tag should be displayed
+  - receive a `hit` as parameter (coming from Algolia)
+- `onUpdate` (function): call each time a tag is added or removed
+  - receive `actualTags` and `oldTags` as parameters, both arrays that respectively contains the current selected tags and the previous ones
+
+It is possible to get encapsulate the `<TagBox />` component in an `<Index />` one if you wish to target different indices from the same InstantSearch instance.
+
+## Examples
+
+* using the `connectAutoComplete()` connector: [CodeSandbox](https://codesandbox.io/s/vyvy30vyy)
+
+### Props
+
+| Name                              | Type      | Use                                                 |
+| --------------------------------- | --------- | --------------------------------------------------- |
+| tagSelectedComponent (required)   | Component | Describe how a selected tag should be displayed.    |
+| tagSuggestionComponent (required) | Component | Describe how a suggested tag should be displayed.   |
+| onUpdate (required)               | Function  | Call each time a tag is added or removed.           |
+| createTagComponent                | Component | Describe how an unexisting tag should be displayed. |
```

Comment:
Not sure about this suggestion but maybe call it something like `noResultsComponent`?

I feel like it would make sense to expose the internal `onTagAdd` and to let the user use it on click on this `noResultsComponent` so that this component is not exclusively used to create a new tag. It would remove the props `createTagAttribute`.